src/main/java/com/github/quadflask/smartcrop/SmartCrop.java

```diff
@@ -9,284 +9,375 @@
+                    crops.add(new Crop(
+                            x,
+                            y,
+                            (int) (minDimension * scale),
+                            (int) (minDimension * scale)));
+                }
+            }
+        }
+        return crops;
+    }
+
+    private Score score(Image output, Crop crop) {
+        Score score = new Score();
+        int[] od = output.getRGB();
+        int width = output.width * options.getScoreDownSample();
+        int height = output.height * options.getScoreDownSample();
+
+        float invDownSample = 1 / options.getScoreDownSample();
+
+        for (int y = 0; y < height; y += options.getScoreDownSample()) {
+            for (int x = 0; x < width; x += options.getScoreDownSample()) {
+                int p = Math.round(y * invDownSample * width + x * invDownSample + 0.5f) * 4;
+                float importance = importance(crop, x, y);
+
+                float detail = (od[p] >> 8 & 0xff) / 255f;
+                score.skin += (od[p] >> 16 & 0xff) / 255f * (detail + options.getSkinBias()) * importance;
+                score.detail += detail * importance;
+                score.saturation += (od[p] & 0xff) / 255f * (detail + options.getSaturationBias()) * importance;
+                score.boost += (od[p] >> 24 & 0xff) / 255f * importance;
+            }
+        }
+
+
+        score.total = (score.detail * options.getDetailWeight()
+                + score.skin * options.getSkinWeight()
+                + score.saturation * options.getSaturationWeight()
+                + score.boost * options.getBoostWeight())
+                / crop.width / crop.height;
+        return score;
+    }
+
+    private float importance(Crop crop, int x, int y) {
+        if (crop.x > x
+                || x >= crop.x + crop.width
+                || crop.y > y
+                || y >= crop.y + crop.height)
+            return options.getOutsideImportance();
+
+        float fx = (float) (x - crop.x) / crop.width;
+        float fy = (float) (y - crop.y) / crop.height;
+        float px = Math.abs(0.5f - fx) * 2;
+        float py = Math.abs(0.5f - fy) * 2;
+        // distance from edg;
+        float dx = Math.max(px - 1.0f + options.getEdgeRadius(), 0);
+        float dy = Math.max(py - 1.0f + options.getEdgeRadius(), 0);
+        float d = (dx * dx + dy * dy) * options.getEdgeWeight();
+        d += (float) (1.4142135f - Math.sqrt(px * px + py * py));
+        if (options.isRuleOfThirds()) {
+            d += (Math.max(0, d + 0.5f) * 1.2f) * (thirds(px) + thirds(py));
+        }
+        return d;
+    }
+
+    static class Image {
+        BufferedImage bufferedImage;
+        int width, height;
+        int[] data;
+
+        public Image(int width, int height) {
+            this.width = width;
+            this.height = height;
+            this.data = new int[width * height];
+            for (int i = 0; i < this.data.length; i++)
+                data[i] = 0xff000000;
```

Comment:
FYI, this is going to set the alpha channel/boost score to 255 for the whole image by default, so you might want to change this to zero.